libkbfs/block_tree.go

```diff
@@ -291,159 +311,214 @@ func (bt *blockTree) getBlocksForOffsetRange(ctx context.Context,
 	prefixOk bool, getDirect bool) (pathsFromRoot [][]parentBlockAndChildIndex,
 	blocks map[BlockPointer]Block, nextBlockOffset Offset,
 	err error) {
-	nextBlockOffset = pblock.FirstOffset()
-	if !pblock.IsIndirect() {
-		// Return a single empty path, under the assumption that the
-		// caller already checked the range for this block.
-		if getDirect {
-			// Return a child map with only this block in it.
-			return [][]parentBlockAndChildIndex{nil},
-				map[BlockPointer]Block{ptr: pblock}, nil, nil
+
+	// Make a queue for tasks and one for results. Queue for results is big to prevent workers from blocking on
+	// sending back their results.
+	jobs := make(chan task, 2)
+	results := make(chan result, 400)
```

Comment:
The results chan is being continuously drained by a separate goroutine which doesn't block on anything else, so I don't think this needs to be big.  In fact, it can probably be a 0-length channel and that would be fine.  Am I missing something though?

Reply:
I think this resulted from my misunderstanding where most of the time was going to be taken. I agree 0-length would be fine.